pkgs/applications/networking/instant-messengers/ripcord/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib, stdenv, fetchurl, writeScript, appimageTools, buildFHSUserEnv, makeFontsConf,
+  gsettings-desktop-schemas, gtk3, twemoji-color-font }:
+
+let
+  pname = "ripcord";
+  version = "0.4.23";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://cancel.fm/dl/Ripcord-${version}-x86_64.AppImage";
+    sha256 = "0395w0pwr1cz8ichcbyrsscmm2p7srgjk4vkqvqgwyx41prm0x2h";
+    name = "${pname}-${version}.AppImage";
+  };
+
+  fontsConf = makeFontsConf {
+    fontDirectories = [
+      twemoji-color-font
+    ];
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+
+in buildFHSUserEnv (appimageTools.defaultFhsEnvArgs // {
+  inherit name;
+
+  targetPkgs = pkgs: appimageTools.defaultFhsEnvArgs.targetPkgs pkgs ++ (with pkgs; [ ]);
+
+  extraBuildCommands = ''
+    # Begin patch to fix https://dev.cancel.fm/tktview?name=d2dc78360c
+    chmod u+w ./etc
+
+    # Realise fonts by one level
+    rm ./etc/fonts && mkdir ./etc/fonts
+
+    # Substitute fonts.conf
+    ln -s ${fontsConf.out} ./etc/fonts/fonts.conf
+  '';
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/Ripcord.desktop $out/share/applications/ripcord.desktop
+    install -m 444 -D ${appimageContents}/Ripcord_Icon.png \
+      $out/share/icons/hicolor/512x512/apps/ripcord.png
+  '';
+
+  profile = ''
+    export LC_ALL=C.UTF-8
+    export XDG_DATA_DIRS=${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}:${gtk3}/share/gsettings-schemas/${gtk3.name}:$XDG_DATA_DIRS
+  '';
```

Comment:
I didn't have intensive time to look at this at first, or why doing this was suggested. But I meant to use wrapGAppsHook somehow. But from looking at this, this is a qt app, I don't know why it was needed at all @cole-h?

Reply:
I think I was misunderstood: I linked hlissner's package just as "prior art" and an idea of what using `appimageTools.wrapType2` would look like, not as a request to copy its contents.

In short, I didn't (mean to) suggest doing that, and you even acknowledged it wasn't necessary. Should probably be removed, because it worked fine in my testing prior to that change.

Reply:
@cole-h Ahh, now I understand you. It wasn't suggested :+1:  Sorry if we confused you @bqv

Reply:
No worries, I've reverted it now